pkg/providers/vsphere/validator.go

```diff
@@ -12,15 +14,51 @@ import (
 	"github.com/aws/eks-anywhere/pkg/types"
 )
 
+const (
+	vsphereRootPath         = "/"
+	vsphereTypeFolder       = "Folder"
+	vsphereTypeNetwork      = "Network"
+	vsphereTypeResourcePool = "ResourcePool"
+	vsphereTypeDatastore    = "Datastore"
+)
+
+//go:embed static/globalPrivs.json
+var vsphereGlobalPrivsFile string
+
+//go:embed static/eksUserPrivs.json
+var vsphereUserPrivsFile string
+
+//go:embed static/adminPrivs.json
+var vsphereAdminPrivsFile string
+
+//go:embed static/cnsDatastorePrivs.json
+var vsphereCnsDatastorePrivsFile string
+
+//go:embed static/cnsSearchAndSpbmPrivs.json
+var vsphereCnsSearchAndSpbmPrivsFile string
+
+//go:embed static/cnsVmPrivs.json
+var vsphereCnsVmPrivsFile string
+
+//go:embed static/readOnlyPrivs.json
+var vsphereReadOnlyPrivs string
+
 type Validator struct {
-	govc      ProviderGovcClient
-	netClient networkutils.NetClient
+	govc             ProviderGovcClient
+	netClient        networkutils.NetClient
+	newVSphereClient NewVSphereClientCallable
 }
 
-func NewValidator(govc ProviderGovcClient, netClient networkutils.NetClient) *Validator {
+func NewValidator(govc ProviderGovcClient, netClient networkutils.NetClient, f NewVSphereClientCallable) *Validator {
```

Comment:
I feel like I might be abusing golang here. I want a way to do the following:
- inject a vsphere client mock when testing
- when not testing, I want to automatically specify a sane default vsphere client

Is there a better way to achieve this?

Reply:
Have a look at this, might be some inspo: https://github.com/aws/eks-anywhere/pull/2887